yocs_math_toolkit/include/yocs_math_toolkit/common.hpp

```diff
@@ -8,6 +8,17 @@
 #ifndef COMMON_HPP_
 #define COMMON_HPP_
 
+// deprecation macros
```

Comment:
The ecl which this library already uses has headers for deprecated code already. Doesn't have the clang version though - I'll add that to the ecl and use them here. One less duplication to maintain...